### PR TITLE
H5 clean warning

### PR DESCRIPTION
**Pull request by kyang2014**
check with sonar cloud

### CODE REVIEW

**Comment by kyang2014** [comment]
https://github.com/OPENDAP/bes/pull/631/checks?check_run_id=6546119283
shows all get passed. Why here it shows "some checks were not successful".

File: modules/hdf5_handler/HDF5CFUtil.cc [hunk 1]
```diff
@@ -1274,25 +1275,28 @@ void HDF5CFUtil::cha_co(string &co,const string & vpath) {
                 if(co_rp_sep_pos[0]==0) {
                     // Obtain the '../' position at co 
                     if(co_rp_sep_pos.size() <var_sep_pos.size()) {
-                    size_t var_prefix_pos=var_sep_pos[var_sep_pos.size()-co_rp_sep_pos.size()-1];
+                        size_t var_prefix_pos=var_sep_pos[var_sep_pos.size()-co_rp_sep_pos.size()-1];
+                        string var_prefix=vpath.substr(1,var_prefix_pos);
+                        string co_suffix = co.substr(co_rp_sep_pos[co_rp_sep_pos.size()-1]+rp_sep.size());
+                        co = var_prefix+co_suffix;
+#if 0
```
Comment (jgallagher59701) [inline modules/hdf5_handler/HDF5CFUtil.cc:1282]: I'm not sure what your plans are for the groups of lines with the #if 0 ... #endif brackets - I usually leave these in for one PR and then remove them to make it easier to read what the code is doing. 

Sometimes there are things we really want to keep longer. Anyway, if you know these are never going to be used, we could remove them now and make the software a little easier to understand.

**Comment by ndp-opendap** [comment]
> https://github.com/OPENDAP/bes/pull/631/checks?check_run_id=6546119283
> shows all get passed. Why here it shows "some checks were not successful".

It looks like you canceled the build. Thus it was "not successful". Builds are only successful when they run to completion without error.

**Review by jgallagher59701 — APPROVED** [review]
This looks like it is ready for a merge.

Thanks, James

File: modules/hdf5_handler/HDF5CF.cc [hunk 2]
```diff
@@ -64,7 +64,7 @@ Var::Var(Var *var)
     dimnameflag = var->dimnameflag;
     coord_attr_add_path = var->coord_attr_add_path;
 
-    for (vector<Attribute*>::iterator ira = var->attrs.begin(); ira != var->attrs.end(); ++ira) {
+    for (auto ira = var->attrs.begin(); ira != var->attrs.end(); ++ira) {
```
[the remark this answers is not in the document]
Comment (kyang2014) [reply modules/hdf5_handler/HDF5CF.cc:67]: Add the information to the TODO in the future. Thanks,

File: modules/hdf5_handler/HDF5CF.h [hunk 3]
```diff
@@ -165,15 +165,15 @@ class Dimension {
...
+        size(dimsize)
     {
     }
 
 private:
     hsize_t size;
-    std::string name;
-    std::string newname;
-    bool unlimited_dim;
+    std::string name ="";
```
[the remark this answers is not in the document]
Comment (kyang2014) [reply modules/hdf5_handler/HDF5CF.h:174]: Thanks, will remove this later.

File: modules/hdf5_handler/HDF5CFUtil.cc (hunk 1 above)
Comment (kyang2014) [reply modules/hdf5_handler/HDF5CFUtil.cc:1282]: These #if 0 #endif better kept for a while. Most of them contains useful information. I will see if I can clean up some in the next round.

**Pull request merged**